README.md

```diff
@@ -237,3 +252,300 @@ We rely on many tools and services for which we are grateful ❤ to their develo
 |        <a href="https://www.argos-ci.com/"><img src="./.github/ArgosCI.svg" width="150"/></a>        | Visual regression system             |
+
+## Utilities API
+
+Several utility functions are provided in the [`util`](./src/util) module for traversing and manipulating the DOM. These may be useful for theme and extension authors when there is a need to modify the HTML structure of the document (e.g. adding additional purely presentational elements such as social sharing buttons). Caution should be taken to not overly modify the content. Only use these functions for things that are not possible with CSS alone.
+
+<!-- prettier-ignore-start -->
+<!-- UTIL-API -->
+### Functions
+
+<dl>
+<dt><a href="#ready">ready(func)</a></dt>
+<dd><p>Register a function to be executed when the DOM is fully loaded.</p>
+</dd>
+<dt><a href="#first">first(elem, selector)</a> ⇒ <code>Element</code> | <code>null</code></dt>
+<dd><p>Select the first element matching a CSS selector.</p>
+</dd>
+<dt><a href="#select">select(elem, selector)</a> ⇒ <code>Array.&lt;Element&gt;</code></dt>
+<dd><p>Select all elements matching a CSS selector.</p>
+</dd>
+<dt><a href="#create">create(spec, ...children)</a> ⇒ <code>Element</code></dt>
+<dd><p>Create a new element.</p>
+</dd>
+<dt><a href="#attr">attr(target, name, value)</a> ⇒ <code>string</code> | <code>null</code> | <code>undefined</code></dt>
+<dd><p>Get or set the value of an attribute on an element.</p>
+</dd>
+<dt><a href="#text">text(target, value)</a> ⇒ <code>string</code> | <code>null</code> | <code>undefined</code></dt>
+<dd><p>Get or set the text content of an element.</p>
+</dd>
+<dt><a href="#append">append(target, ...elems)</a></dt>
+<dd><p>Append new child elements to an element.</p>
+</dd>
+<dt><a href="#prepend">prepend(target, ...elems)</a></dt>
+<dd><p>Prepend new child elements to an element.</p>
+</dd>
+<dt><a href="#before">before(target, ...elems)</a></dt>
+<dd><p>Insert new elements before an element.</p>
+</dd>
+<dt><a href="#after">after(target, ...elems)</a></dt>
+<dd><p>Insert new elements after an element.</p>
+</dd>
+<dt><a href="#replace">replace(target, ...elems)</a></dt>
+<dd><p>Replace an element with a new element.</p>
+</dd>
+<dt><a href="#wrap">wrap(target, elem)</a></dt>
+<dd><p>Wrap an element with a new element.</p>
+</dd>
+</dl>
+
+<a name="ready"></a>
+
+### ready(func)
+Register a function to be executed when the DOM is fully loaded.
+
+**Kind**: global function
+**Detail**: Use this to wrap calls to the DOM selection and manipulation functions
+to be sure that the DOM is ready before working on it.
+
+| Param | Type | Description |
+| --- | --- | --- |
+| func | <code>function</code> | Function to register |
+
+**Example**
+```js
+ready(() => {
+  // Use other DOM manipulation functions here
+})
+```
+<a name="first"></a>
+
+### first(elem, selector) ⇒ <code>Element</code> \| <code>null</code>
```

Comment:
As part of the README refinements, we should indicate in the type signature that the first `elem` argument is optional. Right now it's explained in the examples, but I missed it the first time I saw the documentation.